components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+// Configure the Python environments
+envs {
+    bootstrapDirectory = new File(buildDir, 'bootstrap')
+    envsDirectory = new File(buildDir, 'envs')
+
+    // Setup a miniconda environment.
+    conda "Miniconda3", "Miniconda3-latest", "64"
+}
+
+// Setup the python environment before the build starts.
+
+task createBuildDir {
```

Comment:
It's very surprising to me that you need to actually create `project.buildDir`; that's something Gradle should be doing for you.

https://discuss.gradle.org/t/at-what-point-in-the-gradle-lifecycle-is-the-build-folder-created-and-how-should-a-custom-task-ensure-that-it-has-already-happened/4217/2 suggests that you would be better off declaring an `@OutputFile` for this... which I expect that the Python plugin already does.  If it doesn't, prefer to do that -- in fact, you might just need
```
task createBuildDir {
    @OutputDirectory
    project.buildDir
}
```

Reply:
> It's very surprising to me that you need to actually create `project.buildDir`; that's something Gradle should be doing for you.

I know :( This is a known bug from the plugin, see JetBrains/gradle-python-envs#26

> https://discuss.gradle.org/t/at-what-point-in-the-gradle-lifecycle-is-the-build-folder-created-and-how-should-a-custom-task-ensure-that-it-has-already-happened/4217/2 suggests that you would be better off declaring an `@OutputFile` for this... which I expect that the Python plugin already does. If it doesn't, prefer to do that -- in fact, you might just need
> 
> ```
> task createBuildDir {
>     @OutputDirectory
>     project.buildDir
> }
> ```

This fails to build, it says "@OutputDirectory cannot annotate local fields" (or something like that!). However, I fixed the dependency problem by getting rid of `mustRunAfter`, as you highlighted in the previous comment.